pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/partition/HashCodePartitionFunction.java

```diff
@@ -57,4 +56,8 @@ public int getNumPartitions() {
   public String toString() {
     return NAME;
   }
+
+  private int abs(int n) {
+    return (n == Integer.MIN_VALUE) ? 0 : Math.abs(n);
```

Comment:
this is equivalent to `Math.abs(n) & Integer.MAX_VALUE` but the callers could just move the modulo inside the `Math.abs` to avoid dealing with `Integer.MIN_VALUE`

Reply:
Sounds good. This is the same as `ByteArrayPartitionFunction`, so I'll also change that